pkg/storage/engine/disk_map.go

```diff
@@ -158,8 +160,25 @@ func generateTempStorageID() uint64 {
 // the underlying store. The RocksDBMap instance will have a keyspace prefixed
 // by a unique prefix.
 func NewRocksDBMap(e Engine) *RocksDBMap {
+	return newRocksDBMap(e, false)
+}
+
+// NewRocksDBStore creates a new RocksDBMap with the passed in Engine as the
+// underlying store. The RocksDBMap instance will have a keyspace prefixed
+// by a unique prefix. Unlike NewRocksDBMap, Puts with identical keys will
+// write multiple entries (instead of overwriting previous entries) that
+// will be returned during iteration.
+func NewRocksDBStore(e Engine) *RocksDBMap {
```

Comment:
Why not just make `NewRocksDBMap` take in an `allowDuplicates` bool and have the caller specify this rather than choose between `NewRocksDBMap` and `NewRocksDBStore`? I think it would be much clearer.